airflow-core/src/airflow/ui/src/layouts/Details/Grid/TaskInstancesColumn.tsx

```diff
@@ -20,41 +20,75 @@ import { Box } from "@chakra-ui/react";
 import { useParams } from "react-router-dom";
 
 import type { LightGridTaskInstanceSummary } from "openapi/requests/types.gen";
+import { DagVersionIndicator } from "src/components/ui/VersionIndicator";
 
 import { GridTI } from "./GridTI";
 import type { GridTask } from "./utils";
 
 type Props = {
   readonly depth?: number;
+  readonly hasMixedVersions?: boolean;
   readonly nodes: Array<GridTask>;
   readonly onCellClick?: () => void;
   readonly runId: string;
   readonly taskInstances: Array<LightGridTaskInstanceSummary>;
+  readonly versionDisplayMode?: string;
 };
 
-export const TaskInstancesColumn = ({ nodes, onCellClick, runId, taskInstances }: Props) => {
+export const TaskInstancesColumn = ({
+  hasMixedVersions,
+  nodes,
+  onCellClick,
+  runId,
+  taskInstances,
+  versionDisplayMode,
+}: Props) => {
   const { dagId = "" } = useParams();
 
-  return nodes.map((node) => {
+  const taskInstanceMap = new Map(taskInstances.map((ti) => [ti.task_id, ti]));
```

Comment:
What's the reason to change this to a map and then make `hasVersionChangeFlag` an immediately invoked function expression?